model/task/task_test.go

```diff
@@ -1253,6 +1253,63 @@ func TestBulkInsert(t *testing.T) {
 	}
 }
 
+func TestGetTaskIdBetweenIds(t *testing.T) {
+	assert := assert.New(t)
+	assert.NoError(db.ClearCollections(Collection))
+	displayName := "cool-task-9000"
+	buildVarient := "bv"
+	requester := "r"
+	project := "proj"
+	tasks := []Task{}
+	for i := 1; i <= 20; i++ {
+		task := Task{
+			Id:                  "t" + fmt.Sprint(i),
+			DisplayName:         displayName,
+			BuildVariant:        buildVarient,
+			Requester:           requester,
+			Project:             project,
+			RevisionOrderNumber: i,
+		}
+		assert.NoError(task.Insert())
+		tasks = append(tasks, task)
+	}
+	t10, err := GetMidwayTask(tasks[0], tasks[19])
+	assert.NoError(err)
+	assert.NotNil(t10)
+	assert.Equal(t10.RevisionOrderNumber, 10)
+
+	t5, err := GetMidwayTask(tasks[0], tasks[9])
+	assert.NoError(err)
+	assert.NotNil(t5)
+	assert.Equal(t5.RevisionOrderNumber, 5)
```

Comment:
nit: I missed this in the original review, but the expected value is the 1st argument and the actual value is the 2nd. If the assertion fails, doing it like this will print the expected/actual values flipped and may cause confusion.